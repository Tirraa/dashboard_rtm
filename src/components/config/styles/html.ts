/* v8 ignore start */
// Stryker disable all
import type { CSSProperties } from 'react';

import { DEFAULT_VARIANT_COLOR_SCHEME } from '@/config/themes';

export const HTML_STYLE: CSSProperties = {
  colorScheme: DEFAULT_VARIANT_COLOR_SCHEME
};
// Stryker restore all
/* v8 ignore stop */
