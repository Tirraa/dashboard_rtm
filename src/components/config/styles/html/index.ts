import { DEFAULT_VARIANT_COLOR_SCHEME } from '@/config/themes';
import type { CSSProperties } from 'react';

export const HTML_STYLE: CSSProperties = {
  colorScheme: DEFAULT_VARIANT_COLOR_SCHEME
};
