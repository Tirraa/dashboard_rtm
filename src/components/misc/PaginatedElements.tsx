'use client';

import { ChevronLeftIcon, ChevronRightIcon } from '@heroicons/react/24/solid';
import { FunctionComponent, ReactNode, useState } from 'react';
import ReactPaginate from 'react-paginate';

interface PaginatedElementsProps {
  paginatedElements: ReactNode[];
  elementsPerPage: number;
}

const classList = { className: 'w-10 h-10 flex items-centers justify-center bg-gray-500 rounded-md' };

const PaginatedElements: FunctionComponent<PaginatedElementsProps> = ({ paginatedElements, elementsPerPage }) => {
  const [currentPage, setCurrentPage] = useState<number>(1);
  const pagesAmount = Math.ceil(paginatedElements.length / elementsPerPage);

  if (pagesAmount <= 1) return <>{paginatedElements}</>;

  const startIndex = (currentPage - 1) * elementsPerPage;
  const endIndex = startIndex + elementsPerPage;
  const currentElements = paginatedElements.slice(startIndex, endIndex);

  function handlePageClick(event: { selected: number }) {
    const targetPage = event.selected + 1;
    if (currentPage !== targetPage) setCurrentPage(targetPage);
  }

  return (
    <>
      {currentElements}
      <ReactPaginate
        breakLabel="..."
        previousLabel={
          <span {...classList}>
            <ChevronLeftIcon />
          </span>
        }
        nextLabel={
          <span {...classList}>
            <ChevronRightIcon />
          </span>
        }
        onPageChange={handlePageClick}
        pageRangeDisplayed={3}
        pageCount={pagesAmount}
        renderOnZeroPageCount={null}
        containerClassName="flex items-center justify-center mt-8 mb-4 select-none"
        previousLinkClassName={`flex items-center justify-center rounded-md ${currentPage <= 1 ? 'opacity-50 cursor-default' : ''}`}
        nextLinkClassName={`flex items-center justify-center rounded-md ${currentPage >= pagesAmount ? 'opacity-50 cursor-default' : ''}`}
        pageLinkClassName={'flex items-center justify-center border border-solid border-blue-500 hover:bg-blue-500 p-2 rounded-md'}
        activeClassName="bg-gray-600 rounded-md text-white"
      />
    </>
  );
};

export default PaginatedElements;
