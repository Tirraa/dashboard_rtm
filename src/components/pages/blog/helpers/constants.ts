/* v8 ignore start */
// Stryker disable all

export const TAGS_KEY = 'tags';
export const FILTERS_KEY = 'filter';
export const FIRST_FILTER_INDEX = 0;

// Stryker restore all
/* v8 ignore stop */
