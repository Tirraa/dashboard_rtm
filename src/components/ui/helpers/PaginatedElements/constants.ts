/* v8 ignore start */
// Stryker disable all

export const PAGE_KEY = 'page';
export const FIRST_PAGE_PARAM = 1;
export const MIN_PAGES_AMOUNT = 1;

// Stryker restore all
/* v8 ignore stop */
