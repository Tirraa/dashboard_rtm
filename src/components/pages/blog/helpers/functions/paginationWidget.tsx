import type { MaybeNull } from '@rtm/shared-types/CustomUtilityTypes';
import type { Quantity, Count } from '@rtm/shared-types/Numbers';
import type { AppPath } from '@rtm/shared-types/Next';
import type { ReactElement } from 'react';

import { PaginationEllipsis, PaginationPrevious, PaginationItem, PaginationLink, PaginationNext } from '@/components/ui/Pagination';
import { dispatchClickOnLinkOrButtonFirstChild, createURLSearchParams } from '@rtm/shared-lib/html';
import { FIRST_PAGE_PARAM } from '@/components/ui/helpers/PaginatedElements/constants';
import { DropdownMenuItem } from '@/components/ui/DropdownMenu';
import { cn } from '@/lib/tailwind';
import Link from 'next/link';

const getItemHref = (i: Count, pathname: AppPath, searchParams: URLSearchParams, pageKey: string) =>
  pathname + createURLSearchParams({ [pageKey]: i === FIRST_PAGE_PARAM ? null : i }, searchParams);

const buildDropdownMenu = (dropdownItems: ReactElement[], pageNumberIndicator?: Count, isBottomWidget?: boolean) =>
  // eslint-disable-next-line @typescript-eslint/no-magic-numbers
  dropdownItems.length === 0 ? null : (
    <PaginationEllipsis pageNumberIndicator={pageNumberIndicator} isBottomWidget={isBottomWidget} dropdownItems={dropdownItems} key={'ellipsis'} />
  );

const buildPaginationItem = (i: Count, isActive: boolean, pathname: AppPath, searchParams: URLSearchParams, pageKey: string) => (
  <PaginationItem key={`page-${i}`}>
    <PaginationLink
      className={cn('border-none font-bold transition-opacity', {
        'pointer-events-none underline opacity-50': isActive
      })}
      href={getItemHref(i, pathname, searchParams, pageKey)}
      aria-current={isActive ? 'page' : undefined}
      isActive={isActive}
    >
      {i}
    </PaginationLink>
  </PaginationItem>
);

function buildPaginationItemsForMobile(
  pagesAmount: Quantity,
  currentPage: Count,
  pathname: AppPath,
  searchParams: URLSearchParams,
  pageKey: string
): MaybeNull<ReactElement> {
  const maybeDropdown: MaybeNull<ReactElement> = buildDropdown(pagesAmount, currentPage, pathname, searchParams, pageKey);
  if (maybeDropdown === null) return null;
  return <li key="pagination-dropdown-mobile">{maybeDropdown}</li>;
}

function buildPaginationItemsForDesktopTrivialCase(
  pagesAmount: Quantity,
  currentPage: Count,
  pathname: AppPath,
  searchParams: URLSearchParams,
  pageKey: string
) {
  const elements: ReactElement[] = [];

  for (let i = FIRST_PAGE_PARAM; i <= pagesAmount; i++) {
    const isActive = currentPage === i;
    elements.push(buildPaginationItem(i, isActive, pathname, searchParams, pageKey));
  }

  return elements;
}

function buildPaginationItemsForDesktop(
  pagesAmount: Quantity,
  currentPage: Count,
  pathname: AppPath,
  searchParams: URLSearchParams,
  pageKey: string
) {
  const maxUserInterfaceItemsAmount = 3;
  const minUserInterfaceItemsAmount = 2;

  if (pagesAmount === maxUserInterfaceItemsAmount || pagesAmount === minUserInterfaceItemsAmount) {
    return buildPaginationItemsForDesktopTrivialCase(pagesAmount, currentPage, pathname, searchParams, pageKey);
  }

  const activePageIsLastPage = currentPage === pagesAmount;

  const dropdownItems = [];
  let leftItem: MaybeNull<ReactElement> = null;
  let rightItem: MaybeNull<ReactElement> = null;

  function buildDropdownItem(i: Count, isCurrent: boolean = false) {
    const href = getItemHref(i, pathname, searchParams, pageKey);
    const item = (
      <DropdownMenuItem
        onClick={(event) => dispatchClickOnLinkOrButtonFirstChild(event.target)}
        className={cn('p-0', {
          'opacity-50': isCurrent
        })}
        key={`page-${i}`}
      >
        <Link
          className={cn('block w-full border-none px-2 py-1.5 text-center font-bold')}
          aria-current={isCurrent ? 'page' : undefined}
          title={String(i)}
          href={href}
        >
          {i}
        </Link>
      </DropdownMenuItem>
    );
    return item;
  }

  for (let i = FIRST_PAGE_PARAM; i <= pagesAmount; i++) {
    const isActive = currentPage === i;

    if (i === pagesAmount) {
      rightItem = buildPaginationItem(i, isActive, pathname, searchParams, pageKey);
      if (activePageIsLastPage) {
        const dropdownItem = buildDropdownItem(i, true);
        dropdownItems.push(dropdownItem);
      } else {
        const dropdownItem = buildDropdownItem(i);
        dropdownItems.push(dropdownItem);
      }

      if (!leftItem) {
        leftItem = buildPaginationItem(FIRST_PAGE_PARAM, false, pathname, searchParams, pageKey);
      }
      continue;
    }

    if (isActive) {
      leftItem = buildPaginationItem(i, isActive, pathname, searchParams, pageKey);

      const dropdownItem = buildDropdownItem(i, true);
      dropdownItems.push(dropdownItem);

      continue;
    }

    const dropdownItem = buildDropdownItem(i);
    dropdownItems.push(dropdownItem);
  }

  const dropdown: MaybeNull<ReactElement> = buildDropdownMenu(dropdownItems);
  if (dropdown === null) return [leftItem, rightItem];
  return [leftItem, <li key="pagination-dropdown-desktop">{dropdown}</li>, rightItem];
}

export function buildDropdown(
  pagesAmount: Quantity,
  pageFromUrl: Count,
  pathname: AppPath,
  searchParams: URLSearchParams,
  pageKey: string,
  isBottomWidget?: boolean
) {
  const dropdownItems: ReactElement[] = [];
  // eslint-disable-next-line @typescript-eslint/no-magic-numbers
  if (pagesAmount === 1) return null;

  for (let i = FIRST_PAGE_PARAM; i <= pagesAmount; i++) {
    const isActive = pageFromUrl === i;
    const href = getItemHref(i, pathname, searchParams, pageKey);

    const dropdownItem = (
      <DropdownMenuItem onClick={(event) => dispatchClickOnLinkOrButtonFirstChild(event.target)} className="h-10 p-0" key={`page-${i}`}>
        <Link
          className={cn('flex h-full w-full items-center justify-center border-none px-2 text-center font-bold', {
            'rounded-md bg-primary': isActive
          })}
          aria-current={isActive ? 'page' : undefined}
          title={String(i)}
          href={href}
        >
          {i}
        </Link>
      </DropdownMenuItem>
    );

    dropdownItems.push(dropdownItem);
  }

  return buildDropdownMenu(dropdownItems, pageFromUrl, isBottomWidget);
}

export const doBuildPaginationItems = (
  currentPage: Count,
  pagesAmount: Quantity,
  pathname: AppPath,
  searchParams: URLSearchParams,
  isLargeScreen: boolean,
  pageKey: string
) =>
  isLargeScreen
    ? buildPaginationItemsForDesktop(pagesAmount, currentPage, pathname, searchParams, pageKey)
    : buildPaginationItemsForMobile(pagesAmount, currentPage, pathname, searchParams, pageKey);

export const buildPreviousBtn = (prevBtnPageId: Count, pathname: AppPath, searchParams: URLSearchParams, currentPage: Count, pageKey: string) => {
  const isDisabled = currentPage <= FIRST_PAGE_PARAM;

  return (
    <PaginationPrevious
      href={pathname + createURLSearchParams({ [pageKey]: prevBtnPageId <= FIRST_PAGE_PARAM ? null : prevBtnPageId }, searchParams)}
      className={cn('max-lg:h-10 max-lg:w-10 max-lg:p-0', { 'pointer-events-none opacity-50': isDisabled })}
      aria-disabled={isDisabled}
    />
  );
};

export const buildNextBtn = (
  nextBtnPageId: Count,
  pathname: AppPath,
  searchParams: URLSearchParams,
  currentPage: Count,
  pagesAmount: Quantity,
  pageKey: string
) => {
  const isDisabled = currentPage >= pagesAmount;

  return (
    <PaginationNext
      className={cn('max-lg:h-10 max-lg:w-10 max-lg:p-0', { 'pointer-events-none opacity-50': isDisabled })}
      href={pathname + createURLSearchParams({ [pageKey]: nextBtnPageId }, searchParams)}
      aria-disabled={isDisabled}
    />
  );
};
