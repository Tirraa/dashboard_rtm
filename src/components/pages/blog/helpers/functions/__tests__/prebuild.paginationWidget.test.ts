// eslint-disable-next-line import/no-extraneous-dependencies
import type { ReactElement } from 'react';

import * as dispatchClickOnLinkOrButtonFirstChild from '@rtm/shared-lib/portable/html/dispatchClickOnLinkOrButtonFirstChild';
import { FIRST_PAGE_PARAM } from '@/components/ui/helpers/PaginatedElements/constants';
import { POINTER_EVENTS_NONE_NEEDLE } from '𝕍/needles';
import { describe, expect, it, vi } from 'vitest';

import { doBuildPaginationItems, buildPreviousBtn, buildDropdown, buildNextBtn } from '../paginationWidget';

const pathname = 'pathname';
const searchParams = new URLSearchParams('?foo=bar&bar=bar&baz=baz');
const pageKey = 'page';

describe('buildNextBtn', () => {
  it('should build an active nextBtn, given a pagesAmount bigger than currentPage', () => {
    const currentPage = 1;
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const [nextBtnPageId, pagesAmount] = [currentPage + 1, currentPage + 1];

    const nextBtn = buildNextBtn(nextBtnPageId, pathname, searchParams, currentPage, pagesAmount, pageKey);
    expect(nextBtn.props.href).toBe(`${pathname}?foo=bar&bar=bar&baz=baz&page=2`);
  });

  it('should build a disabled nextBtn, given a currentPage bigger than pagesAmount', () => {
    const pagesAmount = 1;
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const [nextBtnPageId, currentPage] = [pagesAmount + 1, pagesAmount + 1];

    const nextBtn = buildNextBtn(nextBtnPageId, pathname, searchParams, currentPage, pagesAmount, pageKey);
    expect(nextBtn.props.className).includes(POINTER_EVENTS_NONE_NEEDLE);
    expect(nextBtn.props['aria-disabled']).toBe(true);
  });

  it('should build a disabled nextBtn, given a currentPage equal to pagesAmount', () => {
    const pagesAmount = 1;
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const [nextBtnPageId, currentPage] = [pagesAmount + 1, pagesAmount];

    const nextBtn = buildNextBtn(nextBtnPageId, pathname, searchParams, currentPage, pagesAmount, pageKey);
    expect(nextBtn.props.className).includes(POINTER_EVENTS_NONE_NEEDLE);
    expect(nextBtn.props['aria-disabled']).toBe(true);
  });
});

describe('buildPreviousBtn', () => {
  it('should build an active previousBtn with cleaned page param, given a currentPage just bigger than FIRST_PAGE_PARAM', () => {
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const currentPage = FIRST_PAGE_PARAM + 1;
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const prevBtnPageId = currentPage - 1;

    const previousBtn = buildPreviousBtn(prevBtnPageId, pathname, searchParams, currentPage, pageKey);
    expect(previousBtn.props.href).toBe(`${pathname}?foo=bar&bar=bar&baz=baz`);
  });

  it('should build an active previousBtn, given a currentPage 2 steps after FIRST_PAGE_PARAM', () => {
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const currentPage = FIRST_PAGE_PARAM + 2;
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const prevBtnPageId = currentPage - 1;

    const previousBtn = buildPreviousBtn(prevBtnPageId, pathname, searchParams, currentPage, pageKey);
    expect(previousBtn.props.href).toBe(`${pathname}?foo=bar&bar=bar&baz=baz&page=2`);
  });

  it('should build a disabled previousBtn, given a currentPage equal to FIRST_PAGE_PARAM', () => {
    const currentPage = FIRST_PAGE_PARAM;
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const prevBtnPageId = currentPage - 1;

    const previousBtn = buildPreviousBtn(prevBtnPageId, pathname, searchParams, currentPage, pageKey);
    expect(previousBtn.props.className).includes(POINTER_EVENTS_NONE_NEEDLE);
    expect(previousBtn.props['aria-disabled']).toBe(true);
  });

  it('should build a disabled previousBtn, given FIRST_PAGE_PARAM is bigger than currentPage', () => {
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const currentPage = FIRST_PAGE_PARAM - 1;
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    const prevBtnPageId = currentPage - 1;

    const previousBtn = buildPreviousBtn(prevBtnPageId, pathname, searchParams, currentPage, pageKey);
    expect(previousBtn.props.className).includes(POINTER_EVENTS_NONE_NEEDLE);
    expect(previousBtn.props['aria-disabled']).toBe(true);
  });
});

describe('doBuildPaginationItems', () => {
  it('should match snapshot (desktop, trivial pagesAmount === 2 case, i.e: 2 parts UI without dropdown)', () => {
    const currentPage = 1;
    const pagesAmount = 2;
    const isLargeScreen = true;
    const paginationItems = doBuildPaginationItems(currentPage, pagesAmount, pathname, searchParams, isLargeScreen, pageKey);

    expect(paginationItems).toMatchSnapshot();
  });

  it('should match snapshot (desktop, trivial pagesAmount === 3 case, i.e: 3 parts UI without dropdown)', () => {
    const currentPage = 1;
    const pagesAmount = 3;
    const isLargeScreen = true;
    const paginationItems = doBuildPaginationItems(currentPage, pagesAmount, pathname, searchParams, isLargeScreen, pageKey);

    expect(paginationItems).toMatchSnapshot();
  });

  it('should match snapshot (desktop, 3 parts UI with dropdown)', () => {
    const currentPage = 1;
    const pagesAmount = 4;
    const isLargeScreen = true;
    const paginationItems = doBuildPaginationItems(currentPage, pagesAmount, pathname, searchParams, isLargeScreen, pageKey);

    expect(paginationItems).toMatchSnapshot();
  });

  it('should match snapshot (desktop, 3 parts UI with dropdown, current page being last page)', () => {
    const pagesAmount = 4;
    const currentPage = pagesAmount;
    const isLargeScreen = true;
    const paginationItems = doBuildPaginationItems(currentPage, pagesAmount, pathname, searchParams, isLargeScreen, pageKey);

    expect(paginationItems).toMatchSnapshot();

    const dispatchClickOnLinkOrButtonFirstChildSpy = vi.spyOn(dispatchClickOnLinkOrButtonFirstChild, 'default');

    const dummyEventTarget = new EventTarget();
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    (paginationItems as ReactElement[])[1].props.dropdownItems[0].props.onClick(dummyEventTarget);
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    expect(dispatchClickOnLinkOrButtonFirstChildSpy).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot (mobile)', () => {
    const currentPage = 1;
    const pagesAmount = 3;
    const isLargeScreen = false;
    const paginationItems = doBuildPaginationItems(currentPage, pagesAmount, pathname, searchParams, isLargeScreen, pageKey);

    expect(paginationItems).toMatchSnapshot();

    const dispatchClickOnLinkOrButtonFirstChildSpy = vi.spyOn(dispatchClickOnLinkOrButtonFirstChild, 'default');

    const dummyEventTarget = new EventTarget();
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    (paginationItems as ReactElement).props.dropdownItems[0].props.onClick(dummyEventTarget);
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    expect(dispatchClickOnLinkOrButtonFirstChildSpy).toHaveBeenCalledTimes(1);
  });
});

describe('buildDropdown', () => {
  it('should match snapshot (bottom widget)', () => {
    const currentPage = 1;
    const pagesAmount = 4;
    const isBottomWidget = true;
    const dropdown = buildDropdown(pagesAmount, currentPage, pathname, searchParams, pageKey, isBottomWidget);

    expect(dropdown).toMatchSnapshot();
  });

  it('should match snapshot (NOT bottom widget)', () => {
    const currentPage = 1;
    const pagesAmount = 4;
    const isBottomWidget = false;
    const dropdown = buildDropdown(pagesAmount, currentPage, pathname, searchParams, pageKey, isBottomWidget);

    expect(dropdown).toMatchSnapshot();
  });

  it('should return null, given pagesAmount equal to 1', () => {
    const currentPage = 1;
    const pagesAmount = 1;
    const isBottomWidget = false;
    const dropdown = buildDropdown(pagesAmount, currentPage, pathname, searchParams, pageKey, isBottomWidget);

    expect(dropdown).toBe(null);
  });
});
