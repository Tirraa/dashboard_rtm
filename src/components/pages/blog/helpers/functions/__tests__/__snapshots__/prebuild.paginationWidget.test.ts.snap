// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildDropdown > should match snapshot (NOT bottom widget) 1`] = `
<PaginationEllipsis
  dropdownItems={
    [
      <DropdownMenuItem
        className="h-10 p-0"
        onClick={[Function]}
      >
        <ForwardRef(LinkComponent)
          aria-current="page"
          className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold rounded-md bg-primary"
          href="pathname?foo=bar&bar=bar&baz=baz"
          title="1"
        >
          1
        </ForwardRef(LinkComponent)>
      </DropdownMenuItem>,
      <DropdownMenuItem
        className="h-10 p-0"
        onClick={[Function]}
      >
        <ForwardRef(LinkComponent)
          className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold"
          href="pathname?foo=bar&bar=bar&baz=baz&page=2"
          title="2"
        >
          2
        </ForwardRef(LinkComponent)>
      </DropdownMenuItem>,
      <DropdownMenuItem
        className="h-10 p-0"
        onClick={[Function]}
      >
        <ForwardRef(LinkComponent)
          className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold"
          href="pathname?foo=bar&bar=bar&baz=baz&page=3"
          title="3"
        >
          3
        </ForwardRef(LinkComponent)>
      </DropdownMenuItem>,
      <DropdownMenuItem
        className="h-10 p-0"
        onClick={[Function]}
      >
        <ForwardRef(LinkComponent)
          className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold"
          href="pathname?foo=bar&bar=bar&baz=baz&page=4"
          title="4"
        >
          4
        </ForwardRef(LinkComponent)>
      </DropdownMenuItem>,
    ]
  }
  isBottomWidget={false}
  pageNumberIndicator={1}
/>
`;

exports[`buildDropdown > should match snapshot (bottom widget) 1`] = `
<PaginationEllipsis
  dropdownItems={
    [
      <DropdownMenuItem
        className="h-10 p-0"
        onClick={[Function]}
      >
        <ForwardRef(LinkComponent)
          aria-current="page"
          className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold rounded-md bg-primary"
          href="pathname?foo=bar&bar=bar&baz=baz"
          title="1"
        >
          1
        </ForwardRef(LinkComponent)>
      </DropdownMenuItem>,
      <DropdownMenuItem
        className="h-10 p-0"
        onClick={[Function]}
      >
        <ForwardRef(LinkComponent)
          className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold"
          href="pathname?foo=bar&bar=bar&baz=baz&page=2"
          title="2"
        >
          2
        </ForwardRef(LinkComponent)>
      </DropdownMenuItem>,
      <DropdownMenuItem
        className="h-10 p-0"
        onClick={[Function]}
      >
        <ForwardRef(LinkComponent)
          className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold"
          href="pathname?foo=bar&bar=bar&baz=baz&page=3"
          title="3"
        >
          3
        </ForwardRef(LinkComponent)>
      </DropdownMenuItem>,
      <DropdownMenuItem
        className="h-10 p-0"
        onClick={[Function]}
      >
        <ForwardRef(LinkComponent)
          className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold"
          href="pathname?foo=bar&bar=bar&baz=baz&page=4"
          title="4"
        >
          4
        </ForwardRef(LinkComponent)>
      </DropdownMenuItem>,
    ]
  }
  isBottomWidget={true}
  pageNumberIndicator={1}
/>
`;

exports[`doBuildPaginationItems > should match snapshot (desktop, 3 parts UI with dropdown) 1`] = `
[
  <PaginationItem>
    <PaginationLink
      aria-current="page"
      className="border-none font-bold transition-opacity pointer-events-none underline opacity-50"
      href="pathname?foo=bar&bar=bar&baz=baz"
      isActive={true}
    >
      1
    </PaginationLink>
  </PaginationItem>,
  <li>
    <PaginationEllipsis
      dropdownItems={
        [
          <DropdownMenuItem
            className="p-0 opacity-50"
            onClick={[Function]}
          >
            <ForwardRef(LinkComponent)
              aria-current="page"
              className="block w-full border-none px-2 py-1.5 text-center font-bold"
              href="pathname?foo=bar&bar=bar&baz=baz"
              title="1"
            >
              1
            </ForwardRef(LinkComponent)>
          </DropdownMenuItem>,
          <DropdownMenuItem
            className="p-0"
            onClick={[Function]}
          >
            <ForwardRef(LinkComponent)
              className="block w-full border-none px-2 py-1.5 text-center font-bold"
              href="pathname?foo=bar&bar=bar&baz=baz&page=2"
              title="2"
            >
              2
            </ForwardRef(LinkComponent)>
          </DropdownMenuItem>,
          <DropdownMenuItem
            className="p-0"
            onClick={[Function]}
          >
            <ForwardRef(LinkComponent)
              className="block w-full border-none px-2 py-1.5 text-center font-bold"
              href="pathname?foo=bar&bar=bar&baz=baz&page=3"
              title="3"
            >
              3
            </ForwardRef(LinkComponent)>
          </DropdownMenuItem>,
          <DropdownMenuItem
            className="p-0"
            onClick={[Function]}
          >
            <ForwardRef(LinkComponent)
              className="block w-full border-none px-2 py-1.5 text-center font-bold"
              href="pathname?foo=bar&bar=bar&baz=baz&page=4"
              title="4"
            >
              4
            </ForwardRef(LinkComponent)>
          </DropdownMenuItem>,
        ]
      }
    />
  </li>,
  <PaginationItem>
    <PaginationLink
      className="border-none font-bold transition-opacity"
      href="pathname?foo=bar&bar=bar&baz=baz&page=4"
      isActive={false}
    >
      4
    </PaginationLink>
  </PaginationItem>,
]
`;

exports[`doBuildPaginationItems > should match snapshot (desktop, 3 parts UI with dropdown, current page being last page) 1`] = `
[
  <PaginationItem>
    <PaginationLink
      className="border-none font-bold transition-opacity"
      href="pathname?foo=bar&bar=bar&baz=baz"
      isActive={false}
    >
      1
    </PaginationLink>
  </PaginationItem>,
  <li>
    <PaginationEllipsis
      dropdownItems={
        [
          <DropdownMenuItem
            className="p-0"
            onClick={[Function]}
          >
            <ForwardRef(LinkComponent)
              className="block w-full border-none px-2 py-1.5 text-center font-bold"
              href="pathname?foo=bar&bar=bar&baz=baz"
              title="1"
            >
              1
            </ForwardRef(LinkComponent)>
          </DropdownMenuItem>,
          <DropdownMenuItem
            className="p-0"
            onClick={[Function]}
          >
            <ForwardRef(LinkComponent)
              className="block w-full border-none px-2 py-1.5 text-center font-bold"
              href="pathname?foo=bar&bar=bar&baz=baz&page=2"
              title="2"
            >
              2
            </ForwardRef(LinkComponent)>
          </DropdownMenuItem>,
          <DropdownMenuItem
            className="p-0"
            onClick={[Function]}
          >
            <ForwardRef(LinkComponent)
              className="block w-full border-none px-2 py-1.5 text-center font-bold"
              href="pathname?foo=bar&bar=bar&baz=baz&page=3"
              title="3"
            >
              3
            </ForwardRef(LinkComponent)>
          </DropdownMenuItem>,
          <DropdownMenuItem
            className="p-0 opacity-50"
            onClick={[Function]}
          >
            <ForwardRef(LinkComponent)
              aria-current="page"
              className="block w-full border-none px-2 py-1.5 text-center font-bold"
              href="pathname?foo=bar&bar=bar&baz=baz&page=4"
              title="4"
            >
              4
            </ForwardRef(LinkComponent)>
          </DropdownMenuItem>,
        ]
      }
    />
  </li>,
  <PaginationItem>
    <PaginationLink
      aria-current="page"
      className="border-none font-bold transition-opacity pointer-events-none underline opacity-50"
      href="pathname?foo=bar&bar=bar&baz=baz&page=4"
      isActive={true}
    >
      4
    </PaginationLink>
  </PaginationItem>,
]
`;

exports[`doBuildPaginationItems > should match snapshot (desktop, trivial pagesAmount === 2 case, i.e: 2 parts UI without dropdown) 1`] = `
[
  <PaginationItem>
    <PaginationLink
      aria-current="page"
      className="border-none font-bold transition-opacity pointer-events-none underline opacity-50"
      href="pathname?foo=bar&bar=bar&baz=baz"
      isActive={true}
    >
      1
    </PaginationLink>
  </PaginationItem>,
  <PaginationItem>
    <PaginationLink
      className="border-none font-bold transition-opacity"
      href="pathname?foo=bar&bar=bar&baz=baz&page=2"
      isActive={false}
    >
      2
    </PaginationLink>
  </PaginationItem>,
]
`;

exports[`doBuildPaginationItems > should match snapshot (desktop, trivial pagesAmount === 3 case, i.e: 3 parts UI without dropdown) 1`] = `
[
  <PaginationItem>
    <PaginationLink
      aria-current="page"
      className="border-none font-bold transition-opacity pointer-events-none underline opacity-50"
      href="pathname?foo=bar&bar=bar&baz=baz"
      isActive={true}
    >
      1
    </PaginationLink>
  </PaginationItem>,
  <PaginationItem>
    <PaginationLink
      className="border-none font-bold transition-opacity"
      href="pathname?foo=bar&bar=bar&baz=baz&page=2"
      isActive={false}
    >
      2
    </PaginationLink>
  </PaginationItem>,
  <PaginationItem>
    <PaginationLink
      className="border-none font-bold transition-opacity"
      href="pathname?foo=bar&bar=bar&baz=baz&page=3"
      isActive={false}
    >
      3
    </PaginationLink>
  </PaginationItem>,
]
`;

exports[`doBuildPaginationItems > should match snapshot (mobile) 1`] = `
<li>
  <PaginationEllipsis
    dropdownItems={
      [
        <DropdownMenuItem
          className="h-10 p-0"
          onClick={[Function]}
        >
          <ForwardRef(LinkComponent)
            aria-current="page"
            className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold rounded-md bg-primary"
            href="pathname?foo=bar&bar=bar&baz=baz"
            title="1"
          >
            1
          </ForwardRef(LinkComponent)>
        </DropdownMenuItem>,
        <DropdownMenuItem
          className="h-10 p-0"
          onClick={[Function]}
        >
          <ForwardRef(LinkComponent)
            className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold"
            href="pathname?foo=bar&bar=bar&baz=baz&page=2"
            title="2"
          >
            2
          </ForwardRef(LinkComponent)>
        </DropdownMenuItem>,
        <DropdownMenuItem
          className="h-10 p-0"
          onClick={[Function]}
        >
          <ForwardRef(LinkComponent)
            className="flex h-full w-full items-center justify-center border-none px-2 text-center font-bold"
            href="pathname?foo=bar&bar=bar&baz=baz&page=3"
            title="3"
          >
            3
          </ForwardRef(LinkComponent)>
        </DropdownMenuItem>,
      ]
    }
    pageNumberIndicator={1}
  />
</li>
`;
